### PR TITLE
add twig writer and extra options for exporters to customize the the writer

### PR DESCRIPTION
**Pull request by pbergman**

### CODE REVIEW

File: src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php [hunk 1]
```diff
@@ -99,21 +102,230 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    private function getSourceIteratorInter($sql)
+    {
+        return new DoctrineDBALConnectionSourceIterator($this->getConnection(), $sql);
     }
 
     /**
      * @return Connection
      */
-    protected function getConnection()
+    private function getConnection()
```
Comment (muhammedeminakbulut) [inline src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php]: dit is een breaking change :)

Comment (boudewijn-zicht) [reply src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php]: It is a code change in a Command file... strictly speaking you are right, but in this case I would prefer to consistently use private.

**Review by boudewijn-zicht — APPROVED** [review]
If you can add an entry in the changelog, then you are good to go

File: src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php [hunk 2]
```diff
@@ -28,6 +29,7 @@ class ExportSqlCommand extends ContainerAwareCommand
         'xmlexcel'  => 'Exporter\\Writer\\XmlExcelWriter',
         'json'      => 'Exporter\\Writer\\JsonWriter',
         'csv'       => 'Exporter\\Writer\\CsvWriter',
+        'twig'      => 'Zicht\Bundle\AdminBundle\Exporter\Writer\TwigWriter',
```
Comment (7ochem) [inline src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php]: Officially double backslashes `\\` and this is also consistent with previous lines

File: src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php [hunk 3]
```diff
@@ -99,21 +102,230 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    <info>enclosure</info>       Set the enclosure <comment>[default: "]</comment>
+    <info>no-headers</info>      Disable the headers in the csv <comment>[default: true]</comment>
+    <info>no-bom</info>          Disable the bom (utf8) headers <comment>[default: true]</comment>
+    <info>terminate</info>       Set the terminate string <comment>[default: "\\n"]</comment>
+    
+<comment>TWIG</comment>
+    <info>template</info>        Set the output template <comment>[default: "{{ print()|raw }}"]</comment> 
+    <info>no-eol</info>          Disable the extra EOL character after each row
+   
+    the twig instance, will get to extra filter:
```
Comment (7ochem) [inline src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php]: `will get two extra filters`

File: src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php [hunk 4]
```diff
@@ -99,21 +102,230 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    
+<comment>TWIG</comment>
+    <info>template</info>        Set the output template <comment>[default: "{{ print()|raw }}"]</comment> 
+    <info>no-eol</info>          Disable the extra EOL character after each row
+   
+    the twig instance, will get to extra filter: 
+        print       this will print the result row
+        sql_escape  this will do an sql escape in the input
+    
+    to us an template for printing the rows you could use the internal include from twig:
```
Comment (7ochem) [inline src/Zicht/Bundle/AdminBundle/Command/ExportSqlCommand.php]: `to use a template`

**Review by 7ochem — CHANGES_REQUESTED** [review]

File: src/Zicht/Bundle/AdminBundle/Exporter/Writer/TwigWriter.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        $this->filename = $file;
+        $this->template = $template;
+        $this->twig = $twig;
+        $this->addEol = $addEol;
+    }
+
+
+    /**
+     * @inheritdoc
```
Comment (7ochem) [inline src/Zicht/Bundle/AdminBundle/Exporter/Writer/TwigWriter.php]: Should be `{@inheritdoc}`, also with the other methods down below

**Comment by 7ochem** [comment]
> If you can add an entry in the changelog, then you are good to go

Can you still provide an entry to the CHANGELOG.md?

**Review by 7ochem — DISMISSED** [review]

**Pull request merged**